### PR TITLE
Upgrade submodule oneDNN to v3.7

### LINKED ISSUE #136348: [channels_last] Segmentation fault with aten.convolution
### 🐛 Describe the bug

Segmentation fault is observed with aten.convolution for inputs in channels_last format. This is observed with both eager and compile mode.

The example works fine if channels_last format is not enabled.

### Error logs

```
Internal Error: Received signal - Segmentation fault
Segmentation fault (core dumped)
```

### Minified repro

```
import torch

def compute(input, input2):
    clone = torch.ops.aten.clone.default(input2, memory_format = torch.channels_last)
    convolution = torch.ops.aten.convolution.default(clone, input, None, [2, 13], [1, 14], [1, 1], False, [0, 0], 67)
    convolution_backward = torch.ops.aten.convolution_backward.default(convolution, clone, input, [0], [2, 13], [1, 14], [1, 1], False, [0, 0], 67, [True, True, False])
    print(convolution_backward)

input = torch.randn(134, 67, 2, 1)
input2 = torch.randn(18, 4489, 23, 12)
input = input.contiguous(memory_format=torch.channels_last)

compute(input, input2)
```

### Versions

```
PyTorch version: 2.4.0a0+git9cc5232
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: 14.0.5
CMake version: version 3.28.3
Libc version: glibc-2.35

Python version: 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-119-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
```

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @chauhang @penguinwu

### PR DESCRIPTION
**Pull request by yanbing-j**
This PR is to upgrade submodule oneDNN to v3.7.

## Improvements

- Improved performance of convolution and matmul primitives on Intel Xeon processors with Intel AMX instruction set support (formerly Sapphire Rapids and Granite Rapids).
- Improved performance of int8 and fp32 forward convolution primitive on processors with Intel AVX2 instruction set support.
- Improved performance of fp8 matmul primitives with bf16 and fp16 bias data type on Intel Xeon processors with Intel AMX instruction set support (formerly Sapphire Rapids and Granite Rapids).
- Introduced initial optimizations for Intel GPUs based on Xe3 architecture.
- Added bfloat16 support for SDPA, implemented fp16 and bf16 gemm kernel in SDPA.
- Fixed f16 matmul accuracy, the issue of SDPA cannot dispatched to ukernel, bf16/fp16/fp32 conv performance, INT8 Kernel trigger page fault, deconvolution precision issue on complex128 and fp64 and gemm correctness issue in float16 issues.
- Improved bf16 matmul performance with fp32 destination with Arm Compute Library (ACL).
- Improved bf16 to fp32 reorder performance.
- Improved bf16 reorder performance.
- Improved bf16 convolution with ACL.

Fixes https://github.com/pytorch/pytorch/issues/136348.

## Validation results on CPU

1. NLP models accuracy/inference/training
image

image

2. Torchbench cpu userbenchmark inference & training

image

3. Inductor quantization

image

4. Dynamo benchmarks
image
image
image
image
image
image
image
image

## Validation results on XPU
Accuracy is same as baseline. Performance is shown below.
image

## Validation results on ARM
image
image

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @Guobing-Chen @Xia-Weiwen @snadampal @malfet @milpuz01 @aditew01 @nikhil-arm @fadara01 @voznesenskym @penguinwu @EikanWang @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @yf225

### CODE REVIEW

**Review by fadara01 — APPROVED** [review]
Looks good to me!

**Review by mingfeima — APPROVED** [review]
LGTM

**Review by atalman — APPROVED** [review]
lgtm

**Comment by wdvr** [comment]
@pytorchmergebot revert -m 'failing some tests on trunk - see below' -c ghfirst

test/test_mkldnn.py::TestMkldnnCPU::test_mul_cpu [GH job link](https://github.com/pytorch/pytorch/actions/runs/13500466367/job/37719190026) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/576ed1e400d069ec2fff6162f82a71ff0bd81f7c)

**Comment by EikanWang** [comment]
@atalman , this XPU failure is introduced by https://github.com/pytorch/pytorch/pull/147541. The failure was not captured due to the absence of  `ciflow/xpu`. https://github.com/pytorch/pytorch/pull/147785 will fix the issue.

**Comment by yanbing-j** [comment]
@xuhancn I have rebased to main manually.

**Comment by vpirogov** [comment]
Windows CI fail is root caused to MSVC 19.29.30158.0 bug that breaks binary primitive constructor in C++17 mode. PR https://github.com/oneapi-src/oneDNN/pull/2783 with a workaround is in validation. We'll have oneDNN v3.7.1 tagged as soon as it lands.
